sessionctx/variable/sysvar.go

```diff
@@ -1007,6 +997,18 @@ var defaultSysVars = []*SysVar{
 	}},
 	{Scope: ScopeGlobal | ScopeSession, Name: BlockEncryptionMode, Value: "aes-128-ecb"},
 	/* TiDB specific variables */
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBRowFormatVersion, Value: strconv.Itoa(DefTiDBRowFormatV1), Type: TypeUnsigned, MinValue: 1, MaxValue: 2, SetGlobal: func(s *SessionVars, val string) error {
```

Comment:
Could you explain why 'ScopeSession' is introduced? IMHO it might be dangerous to change it in a session.

Reply:
It _is_ already a session scoped variable, it just uses the API incorrectly:

- It has special exception to load into the session in `SkipInit()` which I am trying to refactor out here. This is left over from https://github.com/pingcap/tidb/pull/24836
- It has a SetSession function on the struct.
- Lightning calls vars.SetSystemVar() directly which calls the SetSession function, and skips the validation check to confirm that ScopeSession is permitted.